cmd/ci-secret-bootstrap/main.go

```diff
@@ -414,40 +414,42 @@ func constructSecrets(ctx context.Context, config secretbootstrap.Config, bwClie
 
 func updateSecrets(secretsGetters map[string]coreclientset.SecretsGetter, secretsMap map[string][]*coreapi.Secret, force bool) error {
 	var errs []error
+	operationCase := util.SecretUpdate
+
 	for cluster, secrets := range secretsMap {
 		for _, secret := range secrets {
-			logrus.Debugf("handling secret: %s:%s/%s", cluster, secret.Namespace, secret.Name)
-			secretsGetter := secretsGetters[cluster]
-			if existingSecret, err := secretsGetter.Secrets(secret.Namespace).Get(context.TODO(), secret.Name, meta.GetOptions{}); err == nil {
+			logger := logrus.WithFields(logrus.Fields{"cluster": cluster, "namespace": secret.Namespace, "name": secret.Name, "type": secret.Type})
+			logger.Debug("handling secret")
+
+			secretClient := secretsGetters[cluster].Secrets(secret.Namespace)
+			existingSecret, err := secretClient.Get(context.TODO(), secret.Name, meta.GetOptions{})
+			if err == nil {
 				if secret.Type != existingSecret.Type {
-					errs = append(errs, fmt.Errorf("cannot change secret type from %q to %q (immutable field): %s:%s/%s", existingSecret.Type, secret.Type, cluster, secret.Namespace, secret.Name))
-					continue
+					logrus.WithField("existing-type", existingSecret.Type).Info("immutable field needs to be updated. Will re-create the secret")
```

Comment:
I think what we want to do is much easier expressed by issuing a Delete here and ignoring IsNotFound for that and otherwise keeping everything as-is

Reply:
The whole thing got more complicated so I thought its better to refactor it

Reply:
updated this way.